app/scripts/metamask-controller.js

```diff
@@ -8890,4 +8899,63 @@ export default class MetamaskController extends EventEmitter {
       initRequest,
     });
   }
+
+  /**
+   * Upgrades an account to support EIP-7702 delegation.
+   * Uses shared EIP-7702 utility to avoid code duplication.
+   *
+   * @param {string} address - The account address to upgrade
+   * @param {string} upgradeContractAddress - The contract address to delegate to
+   * @param {number} chainId - The chain ID for the upgrade
+   * @returns {Promise<{transactionHash: string, delegatedTo: string}>}
+   */
+  async upgradeAccount(address, upgradeContractAddress, chainId) {
```

Comment:
It might be worth decomposing these hooks out to avoid bloating metamask-controller.js